### PR TITLE
feat: unit tests with pytest setup

### PR DESCRIPTION
**Pull request by micache**
Configuration for running pytest in a GitHub Actions CI workflow. 
This setup ensures that your tests are automatically run on every push or pull request to your repository.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      - id: check-toml
+      - id: check-case-conflict
+      - id: check-docstring-first
+      - id: check-added-large-files
+      - id: trailing-whitespace
+      - id: detect-aws-credentials
+        args: ["--allow-missing-credentials"]
+      - id: detect-private-key
+      - id: end-of-file-fixer
+      - id: mixed-line-ending
```
Comment (Huy1711) [inline .pre-commit-config.yaml]: This usually make commit conflict -> please delete this line

**Review by Huy1711 — APPROVED** [review]
LGTM

File: .github/workflows/unit_tests.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    - name: Run tests
+      if: env.tests_exist == 'true'
+      run: |
+        pytest
+
+    - name: No tests found
+      if: env.tests_exist == 'false'
+      run: |
+        echo "No tests to run, but CI passed!"
```
Comment (sontungkieu) [inline .github/workflows/unit_tests.yml:40]: good code

**Review by sontungkieu — APPROVED** [review]

File: .github/workflows/unit_tests.yml (hunk 2 above)
Comment (micache) [reply .github/workflows/unit_tests.yml:40]: thanks

**Pull request merged**